integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py

```diff
@@ -79,61 +79,76 @@ def __init__(
         # Store the params for marshalling
         self._collection_name = collection_name
         self._embedding_function = embedding_function
+        self._embedding_func = get_embedding_function(embedding_function, **embedding_function_params)
         self._embedding_function_params = embedding_function_params
         self._distance_function = distance_function
+        self._metadata = metadata
+        self._collection = None
 
         self._persist_path = persist_path
         self._host = host
         self._port = port
+        self._chroma_client = None
+
+    @property
+    def chroma_client(self):
```

Comment:
Given the incoming support for async clients in document stores, this pattern of lazy-initializing is no longer suitable. Please use [this pattern](https://github.com/deepset-ai/haystack-experimental/blob/90d5966edfe88a6801d7d8efba121d802af73188/haystack_experimental/document_stores/opensearch/document_store.py#L160) instead (combine both the client and the collection's initialization in it) and ensure that it's called in every public method.

Reply:
I used suggested pattern but because of lint errors, I had to raise valueError for every public method which looks ugly :/

Reply:
See below.